BlueprintUI/Sources/Layout/Layout.swift

```diff
@@ -33,10 +129,172 @@ public protocol Layout {
+    /// container's subelements.
+    ///
+    /// Implement this method to tell your custom layout container’s parent how much space the
+    /// container needs for a set of subelements, given a size constraint. The parent might call
+    /// this method more than once during a layout pass with different proposed sizes to test the
+    /// flexibility of the container.
+    ///
+    /// In Blueprint, parents ultimately choose the size of their children, so the actual size that
+    /// this container is laid out in may not be a size that was returned from this method.
+    ///
+    /// ## Sizing Rules
+    ///
+    /// For performance reasons, the layout engine may deduce the measurement of your container for
+    /// some constraint values without explicitly calling
+    /// ``sizeThatFits(proposal:subelements:cache:)``. To ensure that the deduced value is correct,
+    /// your layout must follow some ground rules:
+    ///
+    /// 1. **Given one fixed constraint axis, the element's growth along the other axis should be
+    ///    _monotonic_.** That is, an element can grow when given a larger constraint, or shrink
+    ///    when given a smaller constraint, but it should never shrink when given a larger
+    ///    constraint. When growing on one axis, it is OK to shrink along the other axis, such as a
+    ///    block of text that re-flows as the width changes.
+    /// 2. If your element has no intrinsic size along an axis, you can represent that in a couple
+    ///     ways:
+    ///     - You can return a fixed value representing the minimum size for your element. In this
+    ///       approach, a containing element is usually responsible for stretching your element to
+    ///       fill desired space.
+    ///     - You can return a size that entirely fills the constraint proposed. In this approach,
+    ///       you **must** return `.infinity` when the constraint is
+    ///       ``SizeConstraint/Axis/unconstrained``. Otherwise, your behavior would be in violation
+    ///       of rule #1.
+    ///
+    /// - Parameters:
+    ///   - proposal: A size constraint for the container. The container's parent element that calls
+    ///     this method might call the method more than once with different constraints to learn
+    ///     more about the container’s flexibility before choosing a size for placement.
+    ///   - subelements: A collection of proxies that represent the elements that the container
+    ///     arranges. You can use the proxies in the collection to get information about the
+    ///     subelements as you determine how much space the container needs to display them.
+    ///   - cache: Optional storage for calculated data that you can share among the methods of your
+    ///     custom layout container. See ``makeCache(subelements:)-4kmy2`` for details.
+    /// - Returns: A size that indicates how much space the container needs to arrange its
+    ///   subelements.
+    func sizeThatFits(
```

Comment:
We've discussed a few times exposing the `Environment` here; is it worth adding a `context` parameter to this method and the place method so we can add optional parameters without breaking changes going forward?

Reply:
(I don't think we have to do this here, but it's likely nice to do somewhere in this stack)

Reply:
Yeah, that makes sense to me, at least for the `Environment`. Not sure how I feel about further muddying the API with a context type though, since we're modeling after SwiftUI. Do you recall any specific things we talked about putting in there?

Reply:
It's more that: We don't know what we might want in there at some point, so I'd rather not cause a big breaking change to add said things later on. That being said, for the args in the `Layout`, the env is probably fine... But a `context` also feels like a fairly cheap way (a) get us the env, and (b) to avoid a large breaking change to introduce an ultimately optional param only a few layouts need later on